src/cpptools.ts

```diff
@@ -446,10 +454,27 @@ export class CppConfigurationProvider implements cpptools.CustomConfigurationPro
         const targetArchFromToolchains = targetFromToolchains ? parseTargetArch(targetFromToolchains) : undefined;
 
         const normalizedCompilerPath = util.platformNormalizePath(compilerPath);
-        const flags = fileGroup.compileFlags ? [...shlex.split(fileGroup.compileFlags)] : target.compileFlags;
-        const { standard, extraDefinitions, targetArch } = parseCompileFlags(this.cpptoolsVersion, flags, lang);
-        const defines = (fileGroup.defines || target.defines).concat(extraDefinitions);
-        const includePath = fileGroup.includePath ? fileGroup.includePath.map(p => p.path) : target.includePath;
+        const compileCommandFragments = useFragments ? (fileGroup.compileCommandFragments || target.compileCommandFragments) : undefined;
+        const getAsFlags = (fragments?: string[]) => {
+            if (!fragments) {
+                return [];
+            }
+            return [...util.flatMap(fragments, fragment => shlex.split(fragment))];
+        };
+        let flags: string[] = [];
+        let extraDefinitions: string[] = [];
+        let standard: StandardVersion;
+        let targetArch: Architecture;
+        let intelliSenseMode: IntelliSenseMode;
+        let defines = (fileGroup.defines || target.defines || []);
+        if (!useFragments) {
+            // Send the intelliSenseMode and standard only for CppTools API v5 and below.
+            flags = getAsFlags(fileGroup.compileCommandFragments || target.compileCommandFragments);
```

Comment:
this will need to change to get compileFlags instead (once the interface changes)